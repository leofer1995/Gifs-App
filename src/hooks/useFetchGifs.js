import { useEffect, useState } from "react"
import { gifGet } from '../helpers/getGifs.js'


export const useFetchGifs = ( category )=>{
    const [state,setState] = useState({
        data:[],
        loading:true,
    });

    useEffect(()=>{

        gifGet(category)
        .then(imgs=>setState({
            data:imgs,
            loading:false,
        }))

    },[category])



    return state;
}