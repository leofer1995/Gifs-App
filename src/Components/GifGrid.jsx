import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
// import { gifGet } from "../helpers/getGifs";
import GifGridItem from "./GifGridItem";
import PropTypes from 'prop-types';

const GifGrid = ({category})=>{
    
    const {data:images, loading} = useFetchGifs(category)
 
    return (
        <>
            <h3>{category}</h3>
            {loading && <p>Loading...</p>}
            <div className='Card-Grid'>
                {images.map(({id, title, url})=>
                    <GifGridItem 
                    key = {id}
                    title = {title}
                    url = {url}
                    /> 
                    )}
            </div>
            <hr/>
        </>
    )
};
GifGrid.propTypes = {
    category:PropTypes.string.isRequired
}
export default GifGrid;