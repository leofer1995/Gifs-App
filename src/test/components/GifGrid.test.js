import { shallow } from "enzyme";
import '@testing-library/jest-dom'
import GifGrid from "../../Components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs")//fingir llamadas a este archivo

describe('pruebas en componente GifGrid',()=>{
    const category = 'Rick and Morty'
    
    test('deberia renderizar GifGrid',()=>{
        useFetchGifs.mockReturnValue({//finge una llamada al archivo donde esta el useFetchGifs
            data:[],
            loading:true,
        });
        const wrapper = shallow(<GifGrid category={category}/>)
        expect(wrapper).toMatchSnapshot();
    });

    test('deberia mostar items cuando se cargan imagenes useFetchGifs',()=>{
        const gifs = [
        {
            id:'abc',
            url:'https://localhost/xxx/cosa.png',
            title:'cualquier cosa',
        },
        {
            id:'abc',
            url:'https://localhost/xxx/cosa.png',
            title:'cualquier cosa',
        }]
        useFetchGifs.mockReturnValue({
            data:gifs,
            loading:false,
        });

        const wrapper = shallow(<GifGrid category={category}/>);
        // expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false)//valida que no existe el tag p de loading
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
    });
});