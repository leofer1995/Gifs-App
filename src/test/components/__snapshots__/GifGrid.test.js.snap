// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en componente GifGrid deberia renderizar GifGrid 1`] = `
<Fragment>
  <h3>
    Rick and Morty
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="Card-Grid"
  />
  <hr />
</Fragment>
`;
